frontend/src/metabase/browse/components/BrowseModels.tsx

```diff
@@ -1,43 +1,132 @@
+import type { CollectionId } from "metabase-types/api";
 
 import { BROWSE_MODELS_LOCALSTORAGE_KEY } from "../constants";
-import { getCollectionViewPreferences, groupModels } from "../utils";
+import {
+  filterModels,
+  getCollectionViewPreferences,
+  groupModels,
+  type ActualModelFilters,
+} from "../utils";
 
-import { CenteredEmptyState } from "./BrowseApp.styled";
+import {
+  BrowseContainer,
+  BrowseHeader,
+  BrowseMain,
+  BrowseSection,
+  CenteredEmptyState,
+} from "./BrowseApp.styled";
 import { ModelGrid } from "./BrowseModels.styled";
 import { ModelExplanationBanner } from "./ModelExplanationBanner";
 import { ModelGroup } from "./ModelGroup";
 
-export const BrowseModels = ({
-  modelsResult,
+const { availableModelFilters } = PLUGIN_CONTENT_VERIFICATION;
+
+export const BrowseModels = () => {
+  const getInitialModelFilters = useCallback(() => {
+    return _.reduce(
+      availableModelFilters,
+      (acc, filter, filterName) => {
+        const storedFilterStatus = localStorage.getItem(
+          `browseFilters.${filterName}`,
+        );
+        const shouldFilterBeActive =
+          storedFilterStatus === null
+            ? filter.activeByDefault
+            : storedFilterStatus === "on";
+        return {
+          ...acc,
+          [filterName]: shouldFilterBeActive,
+        };
+      },
+      {},
+    );
+  }, []);
+
+  const [actualModelFilters, setActualModelFilters] =
+    useState<ActualModelFilters>({});
+
+  useEffect(() => {
+    const initialModelFilters = getInitialModelFilters();
+    setActualModelFilters(initialModelFilters);
+  }, [getInitialModelFilters, setActualModelFilters]);
+
+  const handleModelFilterChange = useCallback(
+    (modelFilterName: string, active: boolean) => {
+      localStorage.setItem(
+        `browseFilters.${modelFilterName}`,
+        active ? "on" : "off",
+      );
+      setActualModelFilters((prev: ActualModelFilters) => {
+        return { ...prev, [modelFilterName]: active };
+      });
+    },
+    [setActualModelFilters],
+  );
+
+  return (
+    <BrowseContainer>
+      <BrowseHeader>
+        <BrowseSection>
+          <Flex w="100%" direction="row" justify="space-between" align="center">
+            <Title order={1} color="text-dark">
+              <Group spacing="sm">
+                <Icon size={18} color={color("brand")} name="model" />
+                {t`Models`}
+              </Group>
+            </Title>
+            <PLUGIN_CONTENT_VERIFICATION.ModelFilterControls
+              actualModelFilters={actualModelFilters}
+              handleModelFilterChange={handleModelFilterChange}
+            />
+          </Flex>
+        </BrowseSection>
+      </BrowseHeader>
+      <BrowseMain>
+        <BrowseSection>
+          <BrowseModelsBody actualModelFilters={actualModelFilters} />
+        </BrowseSection>
+      </BrowseMain>
+    </BrowseContainer>
+  );
+};
+
+export const BrowseModelsBody = ({
```

Comment:
We generally only have 1 component per file. Can we move this into it's own file?

Reply:
This refactoring will create some rebase headaches for downstream branches so I've done it in a new branch `browse/refactor` added to the end of the stack